app/src/main/java/org/fitchfamily/android/dejavu/BackendService.kt

```diff
@@ -280,8 +281,12 @@ class BackendService : LocationBackendService() {
     @Synchronized
     private fun scanAllSensors() {
         if (emitterCache == null) {
-            if (DEBUG) Log.d(TAG, "scanAllSensors() - emitterCache is null?!?")
-            return
+            if (instance == null) { // this should not be necessary, but better make sure
+                if (DEBUG) Log.d(TAG, "scanAllSensors() - instance is null")
+                return
+            }
+            if (DEBUG) Log.d(TAG, "scanAllSensors() - emitterCache is null: creating")
+            emitterCache = Cache(this)
```

Comment:
Did you actually see any of the issues addressed here?

Reply:
This fix is unnecessary and just kept cache existing because on miui service got onstopped if i switch from the app that was using location to the app that still use location. I still didn't understand what's happening when switching between location consumers and what is the service lifecycle